docs/pages/guides/tree-shaking.mdx

```diff
@@ -300,3 +302,211 @@ const { View, Image } = require('react-native-web');
 </Tab>
 
 </Tabs>
+
+## Unused export removal
+
+> Experimentally available in SDK 52 and greater
+
+In Expo SDK 52, you can experimentally enable support for automatically removing unused imports and exports across modules. This is useful for speeding up native OTA downloads and optimizing web performance where JavaScript must be parsed and executed using a standard JavaScript engine.
+
+Consider the following code:
+
+```js index.js
+import { ArrowUp } from './icons';
+
+export default function Home() {
+  return <ArrowUp />;
+}
+```
+
+```js icons.js
+export function ArrowUp() {
+  /* ... */
+}
+
+export function ArrowDown() {
+  /* ... */
+}
+
+export function ArrowRight() {
+  /* ... */
+}
+
+export function ArrowLeft() {
+  /* ... */
+}
+```
+
+Because only `ArrowUp` is used in `index.js`, the production bundle will remove all other components from `icons.js`.
+
+```js icons.js (Output)
+export function ArrowUp() {
+  /* ... */
+}
+```
+
+This system scales up to automatically optimize all import/export syntax in your app, across all platforms. While this results in smaller bundles, processing JS still requires time and computer memory so avoid importing millions of modules.
+
+- Tree-shaking only works on modules that use `import` and `export` syntax. Files that use `module.exports` and `require` will not be tree-shaken.
+- Adding a Babel plugins such as `@babel/plugin-transform-modules-commonjs` will convert `import`/`export` to CJS and break tree-shaking across your project.
+- Modules that are marked as side-effects will not be removed from the graph.
+- `export * from "..."` will be expanded and optimized unless the export uses `module.exports` or `exports`.
+- Tree-shaking only runs in production bundles.
+- All modules in the Expo SDK are shipped as ESM and can be exhaustively tree-shaken.
+
+## Enabling tree shaking
+
+> Experimentally available in SDK 52 and greater
+
+<Step label="1">
+
+First, enable `experimentalImportSupport` and ensure your app builds and runs as expected. You may experience issues with require cycles or mixing commonjs and ESM imports.
+
+```js metro.config.js
+const { getDefaultConfig } = require('expo/metro-config');
+
+const config = getDefaultConfig(__dirname);
+
+config.transformer.getTransformOptions = async () => ({
+  transform: {
+    experimentalImportSupport: true,
+  },
+});
+
+module.exports = config;
+```
+
+Experimental import support uses Metro's version of the `@babel/plugin-transform-modules-commonjs` plugin. This drastically reduces the number of resolutions and simplifies your output bundle. This feature can be used with `inlineRequires` to further optimize your bundle experimentally.
+
+In the future, we'll enable this feature by default.
```

Comment:
```suggestion
```

I think we can just remove this. I don't think there's any concrete significance as to when this will be enabled by default. It makes this section a bit more future proof and when we remove experimental support, we can just update the section to remove those references.